tests/specs/lemmas.k

```diff
@@ -5,6 +5,34 @@ module LEMMAS
     imports LEMMAS-JAVA
     imports LEMMAS-HASKELL
 
+    rule WS ++ .ByteArray => WS  [simplification]
+    rule .ByteArray ++ WS => WS  [simplification]
+
+    //Byte array store operations sorting. Slices with lower index first. Helps applying rule below.
+    rule MEM [ N := BUF:ByteArray ] [ M := BUF' ] => MEM [ M := BUF' ] [ N := BUF ]
+      requires N >Int M andBool N -Int M >=Int #sizeByteArray(BUF')                                               [simplification]
+
+    rule M [ N := BUF ] [ N := BUF' ] => M [ N := BUF' ] requires #sizeByteArray(BUF) ==Int #sizeByteArray(BUF')  [simplification]
+    //when BUF1 and BUF2 partially overlap and cannot be sorted by sorting rule.
+    rule MEM [ N := BUF1 ] [ M := BUF2:ByteArray ] [ N := BUF3 ] => MEM [ M := BUF2 ] [ N := BUF3 ]
+      requires #sizeByteArray(BUF1) ==Int #sizeByteArray(BUF3)                                                    [simplification]
```

Comment:
This rule is too specific for my liking, let's not add it here. The reason is that what if there are 2 intermediate writes instead? Or 3? etc...

Let's remove it for now.

Reply:
ok